dev-tools/mage/module_changeset.go

```diff
@@ -0,0 +1,132 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mage
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+func DefineModules() {
+	// If MODULE is set in Buildkite pipeline step, skip variable further definition
+	if os.Getenv("MODULE") != "" {
+		return
+	}
+
+	if mg.Verbose() {
+		fmt.Printf("Detecting changes in modules\n")
+	}
+
+	beatPath := os.Getenv("BEAT_PATH")
+
+	var modulePattern = fmt.Sprintf("^%s\\/module\\/([^\\/]+)\\/.*", beatPath)
+
+	moduleRegex, err := regexp.Compile(modulePattern)
+	if err != nil {
+		log.Fatal("failed to compile regex: " + err.Error())
+	}
+
+	modules := map[string]struct{}{}
+	for _, line := range getDiff() {
+		if !isAsciiOrPng(line) {
+			matches := moduleRegex.FindStringSubmatch(line)
+			if len(matches) > 0 {
+				modules[matches[1]] = struct{}{}
+			}
+		}
+	}
+
+	keys := make([]string, len(modules))
+	i := 0
+	for k := range modules {
+		keys[i] = k
+		i++
+	}
+
+	moduleVar := strings.Join(keys, ",")
+
+	if moduleVar != "" {
+		err = os.Setenv("MODULE", moduleVar)
```

Comment:
In Jenkins MODULE was always printed into the log, so I kept same approach in Buildkite.
I have updated `func DefineModules` - it will log either MODULE, either message that no modules changed

Reply:
In Jenkins MODULE was always printed out in log, so I decided to keep that in Buildkite in a same way.
I have updated last lines of `func DefineModules` to always print either MODULE, either message that no changed modules found.